runtime/lua/vscode/api/env.lua

```diff
@@ -0,0 +1,60 @@
+local code = require("vscode.api")
+
+---@class Clipboard
+---@field readText fun(): string
+---@field writeText fun(value: string)
+
+---@class Env
+---@field appHost string
+---@field appName string
+---@field appRoot string
+---@field isNewAppInstall boolean
+---@field isTelemetryEnabled boolean
+---@field language string
+---@field logLevel "Off"|"Trace"|"Debug"|"Info"|"Warning"|"Error"
+---@field machineId string
+---@field remoteName string|nil
+---@field sessionId string
+---@field shell string
+---@field uiKind "Desktop"|"Web"
+---@field uriScheme string
+---@field clipboard Clipboard
+local M = {}
+
+M.clipboard = {
+  readText = function()
+    return code.eval("return await vscode.env.clipboard.readText()")
```

Comment:
This is duplicating nvim's own clipboard api.

Again, duplicating interfaces has these costs:

- confusing for users: the don't know where to look to get features. Since now we expose this part of vscode directly, they'll be less likely to discover other builtin parts of Nvim
- adds cost maintenance cost
- in the case of clipboard, it avoids Nvim's clipboard features/providers, which are expected by plugins and user configs